chopper/lib/src/interceptors/headers_interceptor.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:chopper/src/chain/chain.dart';
+import 'package:chopper/src/interceptors/interceptor.dart';
+import 'package:chopper/src/response.dart';
+import 'package:chopper/src/utils.dart';
+import 'package:meta/meta.dart';
+
+/// {@template HeadersInterceptor}
+/// A [Interceptor] that adds [headers] to every request.
+///
+/// Note that this interceptor will overwrite existing headers having the same
+/// keys as [headers].
+/// {@endtemplate}
+@immutable
+class HeadersInterceptor implements Interceptor {
+  final Map<String, String> headers;
+
+  /// {@macro HeadersInterceptor}
+  const HeadersInterceptor(this.headers);
+
+  @override
+  Future<Response<BodyType>> intercept<BodyType>(Chain<BodyType> chain) async =>
```

Comment:
Shouldn't this be
```dart
@override
FutureOr<Response<BodyType>> intercept<BodyType>(Chain<BodyType> chain) =>
    chain.proceed(
      applyHeaders(
        chain.request,
        headers,
      ),
    );
```
since the parent is also a `FutureOr<Response<BodyType>>`